packages/web-api-scan-runner/src/crawl-runner/crawl-runner.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { GlobalLogger } from 'logger';
+import { inject, injectable } from 'inversify';
+import { Crawler, iocTypes as crawlerIocTypes, SimpleCrawlerRunOptions } from 'accessibility-insights-crawler';
+import { Page } from 'puppeteer';
+import { ScanMetadataConfig } from '../scan-metadata-config';
+
+type CrawlerProvider = () => Promise<Crawler<string[]>>;
+
+@injectable()
+export class CrawlRunner {
+    constructor(
+        @inject(crawlerIocTypes.CrawlerProvider) private readonly getCrawler: CrawlerProvider,
+        @inject(GlobalLogger) private readonly logger: GlobalLogger,
+        @inject(ScanMetadataConfig) scanMetadataConfig: ScanMetadataConfig,
+    ) {
+        const scanMetadata = scanMetadataConfig.getConfig();
+        this.logger.setCommonProperties({ scanId: scanMetadata.id });
```

Comment:
We shouldn't need to set logger properties in this class. GlobalLogger is a singleton per batch worker task, and setCommonProperties is already called in Runner.run(), so the common logger properties will already be set by the time CrawlRunner.run() executes.

Reply:
There's no guarantee the id will be set at the point when the CrawlRunner may need it. The Runner class sets the id, but because the CrawlRunner is actually instantiated first, the CrawlRunner is now the first object to set the id.
This is a bit of a flaw in the design IMO. The logger interface consumed by classes which need to log should not give access to setting the scan id if they aren't intended to set it.
In the end, for the moment, we know that the logger isn't used in the CrawlRunner before the scan id is set. But that would be very easy to overlook if the code were to change.
My current thinking is that I'd rather set the id twice than cause a problem if the code changes. I also think the problem wouldn't be detected by unit tests if the code changes.